src/main/java/org/gridsuite/network/map/dto/mapper/VscConverterStationInfosMapper.java

```diff
@@ -60,4 +70,57 @@ private static VscConverterStationTabInfos toTabInfos(Identifiable<?> identifiab
         return builder.build();
     }
 
+    public static VscConverterStationFormInfos toFormInfos(Identifiable<?> identifiable) {
+        VscConverterStation vscConverterStation = (VscConverterStation) identifiable;
+        Terminal terminal = vscConverterStation.getTerminal();
+        VscConverterStationFormInfos.VscConverterStationFormInfosBuilder<?, ?> builder = VscConverterStationFormInfos.builder()
+                .name(vscConverterStation.getOptionalName().orElse(null))
+                .id(vscConverterStation.getId())
+                .voltageLevelId(terminal.getVoltageLevel().getId())
+                .nominalVoltage(terminal.getVoltageLevel().getNominalV())
+                .terminalConnected(terminal.isConnected())
+                .lossFactor(vscConverterStation.getLossFactor())
+                .voltageRegulatorOn(vscConverterStation.isVoltageRegulatorOn());
+
+        if (!Double.isNaN(vscConverterStation.getVoltageSetpoint())) {
+            builder.voltageSetpoint(vscConverterStation.getVoltageSetpoint());
+        }
+
+        if (!Double.isNaN(vscConverterStation.getReactivePowerSetpoint())) {
+            builder.reactivePowerSetpoint(vscConverterStation.getReactivePowerSetpoint());
+        }
+
+        if (!Double.isNaN(terminal.getQ())) {
+            builder.q(terminal.getQ());
+        }
+
+        if (!Double.isNaN(terminal.getP())) {
+            builder.p(terminal.getP());
+        }
+
+        ConnectablePosition<VscConverterStation> connectablePosition = vscConverterStation.getExtension(ConnectablePosition.class);
```

Comment:
You can set the connection infos in just one line : 

builder.connectablePosition(toMapConnectablePosition(vscConverterStation, 0));

after having used the class ConnectablePositionInfos in VscConverterStationFormInfos class (see above ...)

Reply:
done